docs/reference/cheat-sheet.md

```diff
@@ -0,0 +1,452 @@
+```
+
+Response:
+
+```json
+{
+  "flags": [
+    {"key": "simple-boolean", "reason": "STATIC", "variant": "on", "value": true, "metadata": {}},
+    {"key": "simple-string", "reason": "STATIC", "variant": "greeting", "value": "Hello, World!", "metadata": {}},
+    {"key": "simple-number", "reason": "STATIC", "variant": "medium", "value": 50, "metadata": {}}
+  ]
+}
+```
+
+---
+
+## Context-Aware Evaluation
+
+### Context from Request Body
+
+Pass evaluation context in the request body to trigger targeting rules:
+
+```shell
+# Evaluate with email context (triggers email-based-feature targeting)
+curl -X POST 'http://localhost:8016/ofrep/v1/evaluate/flags/email-based-feature' \
+  -H 'Content-Type: application/json' \
+  -d '{"context": {"email": "user@example.com"}}'
+```
+
+Response (email matches `@example.com`):
+
+```json
+{
+  "key": "email-based-feature",
+  "reason": "TARGETING_MATCH",
+  "variant": "on",
+  "value": true,
+  "metadata": {}
+}
+```
+
+```shell
+# Evaluate with tier context
+curl -X POST 'http://localhost:8016/ofrep/v1/evaluate/flags/user-tier-flag' \
+  -H 'Content-Type: application/json' \
+  -d '{"context": {"tier": "premium"}}'
+```
+
+```shell
+# Bulk evaluation with multiple context values
+curl -X POST 'http://localhost:8016/ofrep/v1/evaluate/flags' \
+  -H 'Content-Type: application/json' \
+  -d '{"context": {"email": "admin@example.com", "tier": "enterprise", "region": "us"}}'
+```
+
+### Context from Static Values
+
+Add static context values using the `-X` flag at startup. These are automatically included in all evaluations:
+
+```shell
+docker run --rm -it \
+  -p 8013:8013 -p 8015:8015 -p 8016:8016 \
+  -v $(pwd):/flags \
+  ghcr.io/open-feature/flagd:latest start \
+  --uri file:./flags/cheat-sheet-flags.json \
+  -X region=eu \
+  -X environment=production
+```
+
+```shell
+# region=eu is automatically applied without needing to send context
+curl -X POST 'http://localhost:8016/ofrep/v1/evaluate/flags/region-config'
+```
+
+### Context from HTTP Headers
+
+Map HTTP headers to evaluation context keys using the `-H` flag at startup:
+
+```shell
+docker run --rm -it \
+  -p 8013:8013 -p 8015:8015 -p 8016:8016 \
+  -v $(pwd):/flags \
+  ghcr.io/open-feature/flagd:latest start \
+  --uri file:./flags/cheat-sheet-flags.json \
+  -H "X-User-Tier=tier" \
+  -H "X-User-Email=email"
+```
+
+Now context is extracted from request headers:
+
+```shell
+# tier context comes from X-User-Tier header
+curl -X POST 'http://localhost:8016/ofrep/v1/evaluate/flags/user-tier-flag' \
+  -H 'X-User-Tier: enterprise'
+
+# email context comes from X-User-Email header
+curl -X POST 'http://localhost:8016/ofrep/v1/evaluate/flags/email-based-feature' \
+  -H 'X-User-Email: developer@example.com'
+```
+
+### Context Priority
+
+When using multiple context sources, values are merged with this priority (highest to lowest):
+
+1. Header-mapped context values (`-H` flag)
+2. Static context values (`-X` flag)
+3. Request body context
```

Comment:
Is this mentioned elsewhere in the doc? It's fine to include here but I'd also expect it to be explained in a dedicated evaluation context section.

Reply:
Good call. It's not. I will add a section about it.

Reply:
Added: https://github.com/open-feature/flagd/pull/1839/changes/9d1fca1f58eca8652db3a1a593eb81e8acae2947